gengokit/httptransport/embeddable_funcs.go

```diff
@@ -22,6 +24,15 @@ func PathParams(url string, urlTmpl string) (map[string]string, error) {
 	rv := map[string]string{}
 	pmp := BuildParamMap(urlTmpl)
 
+	tmplLen := len(strings.Split(strings.TrimRight(urlTmpl, "/"), "/"))
+	partsLen := len(strings.Split(strings.TrimRight(url, "/"), "/"))
+	if tmplLen != partsLen {
+		expected := tmplLen
```

Comment:
This seems unnecessary, redeclaring these when you already have them above. I would change the names above to `expectedLen` and `receivedLen`.

Nit: The `Sprintf` below could be using `%d` rather than `%v`.